lib/src/_client.dart

```diff
@@ -306,9 +307,10 @@ class Client {
       if (trackTotalHits != null) 'track_total_hits': trackTotalHits,
       if (size != null) 'size': size,
       if (minScore != null) 'min_score': minScore,
+      if (knn != null) 'knn': knn,
     };
     final params = {
-      'search_type': 'dfs_query_then_fetch',
+      if (knn == null) 'search_type': 'dfs_query_then_fetch',
```

Comment:
Please add a note here why `knn` removes this parameter

Reply:
When I ran without this, it crashed with an error message that said "you can't specify search_type if knn is specified"